tests/components/modbus/test_init.py

```diff
@@ -163,184 +155,115 @@ async def _config_helper(hass, do_config, caplog):
         },
         {
             CONF_TYPE: "tcp",
-            CONF_HOST: "modbusTestHost",
+            CONF_HOST: TEST_HOST,
             CONF_PORT: 5501,
             CONF_DELAY: 5,
         },
+        [
+            {
+                CONF_TYPE: "tcp",
+                CONF_HOST: TEST_HOST,
+                CONF_PORT: 5501,
+                CONF_NAME: TEST_MODBUS_NAME,
+            },
+            {
+                CONF_TYPE: "tcp",
+                CONF_HOST: TEST_HOST,
+                CONF_PORT: 5501,
+                CONF_NAME: TEST_MODBUS_NAME + "2",
+            },
+            {
+                CONF_TYPE: "serial",
+                CONF_BAUDRATE: 9600,
+                CONF_BYTESIZE: 8,
+                CONF_METHOD: "rtu",
+                CONF_PORT: "usb01",
+                CONF_PARITY: "E",
+                CONF_STOPBITS: 1,
+                CONF_NAME: TEST_MODBUS_NAME + "3",
+            },
+        ],
     ],
 )
 async def test_config_modbus(hass, caplog, do_config, mock_pymodbus):
-    """Run test for modbus."""
-    await _config_helper(hass, do_config, caplog)
+    """Run configuration test for modbus."""
+    config = {DOMAIN: do_config}
+    caplog.set_level(logging.ERROR)
+    assert await async_setup_component(hass, DOMAIN, config) is True
+    await hass.async_block_till_done()
+    assert DOMAIN in hass.config.components
+    assert len(caplog.records) == 0
+
 
+async def test_pb_service_write(hass, caplog, mock_modbus):
+    """Run test for service write_register."""
+
+    VALUE = "value"
```

Comment:
We could make these parameters and use `pytest.mark.parametrize`. Probably use `getattr` to access the correct attribute on `mock_modbus`.

Reply:
I did it like this in order to avoid setting up the component x times. But it is changed. and I avoided using `getattr` 😀